### PR TITLE
[Adhoc:agathon-cassandra] Stopping cassandra from using a seed if the…

### PR DESCRIPTION
**Pull request by akonkol**
… seed is itself

### CODE REVIEW

File: agathon-cassandra/src/main/java/com/brighttag/agathon/cassandra/AgathonSeedProvider.java [hunk 1]
```diff
@@ -52,13 +53,17 @@
...
   }
 
-  @VisibleForTesting AgathonSeedProvider(AgathonConnector connector, String ringName) {
+  @VisibleForTesting AgathonSeedProvider(AgathonConnector connector, String ringName, @Nullable InetAddress localAddress) {
     LOG.info("Using AgathonSeedProvider!");
     this.connector = connector;
     this.ringName = ringName;
+    this.localHost = localAddress != null ? localAddress.getHostAddress() : null;
+    if (localHost == null) {
+      LOG.info("Could not determine local ip address ");
```
Comment (codyaray) [inline agathon-cassandra/src/main/java/com/brighttag/agathon/cassandra/AgathonSeedProvider.java]: don't need this check/log, you already log out "Unable to obtain local ip address" in line 118.

File: agathon-cassandra/src/main/java/com/brighttag/agathon/cassandra/AgathonConnector.java [hunk 2]
```diff
@@ -75,6 +76,9 @@ public AgathonConnector(String host, @Nullable Integer port) {
       }
       String data = readFrom(connection);
       LOG.info("Calling Agathon API {} returns {}", url, data);
+      InetAddress localHost = InetAddress.getLocalHost();
+      String localHostIp = localHost.getHostAddress();
+      LOG.info("Local ip is: {} ignoring if in seed list", localHostIp);
```
Comment (codyaray) [inline agathon-cassandra/src/main/java/com/brighttag/agathon/cassandra/AgathonConnector.java]: don't need this, you already log out if ignoring in the provider itself.

File: agathon-cassandra/src/main/java/com/brighttag/agathon/cassandra/AgathonSeedProvider.java [hunk 3]
```diff
@@ -52,13 +53,17 @@
...
-        params.get("ring_name"));
+        params.get("ring_name"), getLocalAddress());
   }
 
-  @VisibleForTesting AgathonSeedProvider(AgathonConnector connector, String ringName) {
+  @VisibleForTesting AgathonSeedProvider(AgathonConnector connector, String ringName, @Nullable InetAddress localAddress) {
     LOG.info("Using AgathonSeedProvider!");
     this.connector = connector;
     this.ringName = ringName;
+    this.localHost = localAddress != null ? localAddress.getHostAddress() : null;
```
Comment (codyaray) [inline agathon-cassandra/src/main/java/com/brighttag/agathon/cassandra/AgathonSeedProvider.java]: instead of calling getHostAddress with the null-checking stuff here, you could make your method return localAddress.getHostAddress() directly. then this can be a simple assignment.

File: agathon-cassandra/src/test/java/com/brighttag/agathon/cassandra/AgathonSeedProviderTest.java [hunk 4]
```diff
@@ -59,34 +63,54 @@ public void tearDown() {
...
     InetAddress addr3 = createMock(InetAddress.class);
+    expect(addr1.getHostAddress()).andReturn("123");
+    expect(addr2.getHostAddress()).andReturn("456");
+    expect(addr3.getHostAddress()).andReturn("789");
+
     expect(connector.getSeeds(RING_NAME)).andReturn(ImmutableList.of(SEED1, SEED2, SEED3));
     expect(provider.getInetAddress(SEED1)).andReturn(addr1);
     expect(provider.getInetAddress(SEED2)).andReturn(addr2);
     expect(provider.getInetAddress(SEED3)).andReturn(addr3);
-    replayAll();
```
Comment (codyaray) [inline agathon-cassandra/src/test/java/com/brighttag/agathon/cassandra/AgathonSeedProviderTest.java]: think I'm missing something... why did you need to replay each mock individually? I see you added replay(localAddr) in setUp(). Assuming that's because you called getHostAddress() in the constructor, my suggestions above should fix that so you can simplify this again.

(General rule of thumb: work in constructors isn't a great idea.)

**Comment by codyaray** [comment]
I like your meta description of this change. "seed is itself"

**Comment by bkochendorfer** [comment]
:shipit:
